src/messages/responses/topic-subscribe.ts

```diff
@@ -45,14 +45,52 @@ export class Item extends Response {
   }
 }
 
+/**
+ * Encapsulates a topic subscription.
+ *
+ * @remarks Currently allows unsubscribing from the topic.
+ * In the future, this may be extended to include additional
+ * statistics about the subscription.
+ */
+export class Subscription extends Response {
+  protected _unsubscribeFn: () => void;
+  private _isSubscribed: boolean;
+
+  constructor(_unsubscribeFn: () => void) {
+    super();
+    this._unsubscribeFn = _unsubscribeFn;
+    this._isSubscribed = true;
+  }
+
+  /**
+   * Unsubscribes from the topic.
+   *
+   * @returns void
+   */
+  public unsubscribe(): void {
+    if (this.isSubscribed()) {
+      this._unsubscribeFn();
+      this._isSubscribed = false;
+    }
+  }
+
+  public setUnsubscribeFn(unsubscribeFn: () => void): void {
+    this._unsubscribeFn = unsubscribeFn;
+  }
```

Comment:
this is the only part of this that gives me slight pause.  I feel like we should be able to pass the constructor to this class a reference to an object, rather than passing it a function.  Then we can mutate that (private, internal) object from outside of this class, without having to expose this function.

Another option would be to make `Subscription` just be an interface rather than a class, and just don't include this mutator on the interface.  But I'm not sure how that would fit into the response class hierarchy; so probably better to just explore the first option that I suggested.